services/dynamic-sidecar/tests/unit/test_models_shared_store.py

```diff
@@ -105,3 +109,18 @@ async def replace_list_in_shared_store(item: str):
         *(replace_list_in_shared_store(f"{x}") for x in range(PARALLEL_CHANGES))
     )
     assert len(shared_store.container_names) == PARALLEL_CHANGES
+
+
+async def test_data_format():
+    MOCKS_DIR = CURRENT_DIR / ".." / "mocks"
+    LEGACY_SHARED_STORE = "legacy_shared_store.json"
+
+    results = await SharedStore.init_from_disk(
+        MOCKS_DIR, store_file_name=LEGACY_SHARED_STORE
+    )
+    # if file is missing it correctly loaded the storage_file
+    assert (MOCKS_DIR / STORE_FILE_NAME).exists() is False
+
+    assert json.loads(results.json()) == json.loads(
```

Comment:
rather than this isolated test, I would rather enforce the serialization function in some base config such that  `SharedStore.Config. json_loads ` is the right one. See https://docs.pydantic.dev/1.10/usage/exporting_models/#custom-json-deserialisation

Reply:
This is just an extra check. it is not checking the serialisation. Here I am checking that the correct content/target to load data from is loaded and correctly parsed.

Reply:
sorry, i think you are testing python here :-(